packages/salesforce/src/util.js

```diff
@@ -1,15 +1,13 @@
 import { Connection } from '@jsforce/jsforce-node';
 import { throwError } from '@openfn/language-common/util';
 
-let connection = null;
-
-async function basicAuth(state) {
+export const basicAuth = async configuration => {
   const { loginUrl, username, password, securityToken, apiVersion } =
-    state.configuration;
+    configuration;
 
-  connection = new Connection({ loginUrl, version: apiVersion });
+  const conn = new Connection({ loginUrl, version: apiVersion });
```

Comment:
I'm not really sure about this design

First off, `basicAuth` doesn't sound like a function that'll create a  Salesforce connection.

Secondly, I was surprised to see new Connection in utils at all.

I would simply have a `connect(configuration)` function which looks at config, creates the right sort of connection. and throws with a clear message if invalid configuration is provided.

The `connect` function could live in Adaptor or Utils, both are fine.

Reply:
It's kind of telling that if you look at `createConnection` in adpator.js,  the only lines of code that actually do anything are calling into `util.js`. So I think these functions all need to be in the same file.

Reply:
I have moved all function for handling connection in Adaptor.js